app/views/foreman/unattended/userdata-katello.erb

```diff
@@ -16,6 +16,9 @@ manage_etc_hosts: true
 ssh_authorized_keys:
```

Comment:
Look at at the line above

```
 <% if @host.params['sshkey'] -%>
 ssh_authorized_keys:
```

This won't render if the user only has remote execution keys, so the YAML will be invalid.